auto-discovery/kubernetes/pull-secret-extractor/internal/secret_extraction/secret_extraction.go

```diff
@@ -0,0 +1,287 @@
+	}, nil
+}
+
+// NewSecretManagerWithOptions creates a SecretManager with custom options (useful for testing)
+func NewSecretManagerWithOptions(client kubernetes.Interface, secretsPath, namespace, podName string, reader SecretReader, creator SecretCreator) *SecretManager {
+	return &SecretManager{
+		k8sClient:     client,
+		secretsPath:   secretsPath,
+		namespace:     namespace,
+		podName:       podName,
+		secretReader:  reader,
+		secretCreator: creator,
+	}
+}
+
+func (sm *SecretManager) CreateTemporarySecret(ctx context.Context, temporarySecretName, domain string) error {
+	if temporarySecretName == "" {
+		return fmt.Errorf("temporary secret name cannot be empty")
+	}
+
+	if domain == "" {
+		return fmt.Errorf("domain cannot be empty")
+	}
+
+	configs, err := sm.secretReader.ReadDockerConfigs(sm.secretsPath)
+	if err != nil {
+		return fmt.Errorf("failed to read Docker configs: %w", err)
+	}
+
+	authEntry := sm.findAuthForDomain(domain, configs)
+	if authEntry == nil {
+		return fmt.Errorf("no authentication found for domain: %s", domain)
+	}
+
+	creds, err := sm.extractCredentials(authEntry)
+	if err != nil {
+		return fmt.Errorf("failed to extract credentials for domain %s: %w", domain, err)
+	}
+
+	secret, err := sm.buildSecret(ctx, temporarySecretName, creds)
+	if err != nil {
+		return fmt.Errorf("failed to build secret: %w", err)
+	}
+
+	if err := sm.secretCreator.CreateSecret(ctx, secret); err != nil {
+		return fmt.Errorf("failed to create temporary secret: %w", err)
+	}
+
+	return nil
+}
+
+func (sm *SecretManager) findAuthForDomain(domain string, configs []DockerConfigJSON) *AuthEntry {
+	for _, config := range configs {
+		if auth, exists := config.Auths[domain]; exists {
+			return &auth
+		}
+	}
+	return nil
+}
+
+func (sm *SecretManager) extractCredentials(auth *AuthEntry) (*Credentials, error) {
+	if auth == nil {
+		return nil, fmt.Errorf("auth entry is nil")
+	}
+
+	if auth.Auth != "" {
+		decoded, err := base64.StdEncoding.DecodeString(auth.Auth)
+		if err != nil {
+			return nil, fmt.Errorf("failed to decode auth field: %w", err)
+		}
+
+		parts := strings.SplitN(string(decoded), ":", 2)
+		if len(parts) != 2 {
+			return nil, fmt.Errorf("invalid auth format, expected username:password")
+		}
+
+		return &Credentials{
+			Username: base64.StdEncoding.EncodeToString([]byte(parts[0])),
+			Password: base64.StdEncoding.EncodeToString([]byte(parts[1])),
+		}, nil
+	}
+
+	if auth.Username != "" && auth.Password != "" {
+		return &Credentials{
+			Username: auth.Username,
+			Password: auth.Password,
+		}, nil
+	}
+
+	return nil, fmt.Errorf("auth entry does not contain valid credentials")
+}
+
+func (sm *SecretManager) buildSecret(ctx context.Context, secretName string, creds *Credentials) (*v1.Secret, error) {
+	pod, err := sm.k8sClient.CoreV1().Pods(sm.namespace).Get(ctx, sm.podName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get pod %s: %w", sm.podName, err)
+	}
```

Comment:
bit of a unfair question, probably something I should rather know, but why do we pass the podname and namespace into the pullSecretExtractor, why not pass in the pod uid?
Then we wouldn't need this lookup. Feels like we are just making our live harder by passing in only the podname/namespace.